pasqal_cloud/utils/retry.py

```diff
@@ -0,0 +1,60 @@
+import functools
+import time
+from typing import (
+    Callable,
+    Iterable,
+    Optional,
+    ParamSpec,
+    TypeVar,
+)
+
+from requests import HTTPError
+
+Param = ParamSpec("Param")
+RT = TypeVar("RT")  # return type
+
+
+def retry_http_error(
+    max_retries: int = 5, retry_status_code: Optional[Iterable[int]] = None
```

Comment:
If possible are we able to begin typing Optional with  `N | None`

Reply:
N | None syntax is only supported in 3.10